caikit/core/model_management/model_finder_base.py

```diff
@@ -0,0 +1,51 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A ModelFinder is responsible for finding models based on a unique input string
+which may be a path, id, or other identifier.
+"""
+
+# Standard
+from typing import Optional
+import abc
+
+# Local
+from ..modules import ModuleConfig
+from ..toolkit.factory import FactoryConstructible
+
+
+class ModelFinderBase(FactoryConstructible):
+    __doc__ = __doc__
```

Comment:
Since its not part of other docs, I was wondering if we should add information about how to initialize and use both finders and initializer in docstrings somewhere, so that it becomes easier to figure out how to use it.

Reply:
That's a good call. I'll add this to the docstrings for the individual types